assets/playground.html

```diff
@@ -948,6 +1082,81 @@
           this.images.push(...urls);
         },
 
+        handleKeyDown(event) {
+          const isMac = navigator.platform.toUpperCase().indexOf('MAC') > -1;
+          const controlKey = isMac ? event.metaKey : event.ctrlKey;
+          if (controlKey && event.shiftKey && event.key.toLowerCase() === 'o') {
+            event.preventDefault();
+            this.handleNewChat();
+          } else if (controlKey && event.shiftKey && event.key.toLowerCase() === 'l') {
+            event.preventDefault();
+            this.showModal = 'list-sessions'
+          } else if (event.shiftKey && event.key === "Escape") {
+            event.preventDefault();
+            this.focusInput();
+          } else if (this.showModal && event.key === "Escape") {
+            event.preventDefault();
+            this.showModal = "";
+          }
+        },
+
+        handleNewChat() {
+          if (this.asking) {
+            this.askAbortController?.abort();
+          }
+          if (this.sessionTitle) {
+            const lastMessage = this.messages[this.messages.length - 1];
+            if (lastMessage.state === "loading") {
+              lastMessage.state = "failed";
+              lastMessage.error = "Error: Aborted";
+              lastMessage.html = renderMarkdown(lastMessage.content, lastMessage.error);
+            }
+            const sessionData = JSON.parse(JSON.stringify({
+              settings: this.settings,
+              messages: this.messages,
+              sessionMode: this.sessionMode,
+              sessionTitle: this.sessionTitle,
+            }));
+            let session = this.sessions.find(v => v.id === this.selectSessionId);
+            if (session) {
+              Object.assign(session, sessionData);
+            } else {
+              this.sessions.unshift({
+                id: randomUUID(),
```

Comment:
could be replaced with [self.crypto.randomUUID()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID#:~:text=self.crypto.randomUUID()%3B)